src/main/java/com/dev/cinema/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,33 @@
+package com.dev.cinema.service.impl;
+
+import com.dev.cinema.dao.UserDao;
+import com.dev.cinema.lib.Inject;
+import com.dev.cinema.lib.Service;
+import com.dev.cinema.model.User;
+import com.dev.cinema.service.UserService;
+import com.dev.cinema.util.HashUtil;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class UserServiceImpl implements UserService {
+    @Inject
+    private UserDao userDao;
+
+    @Override
+    public User add(User user) {
+        user.setSalt(HashUtil.getSalt());
+        user.setPassword(HashUtil.hashPassword(user.getPassword(), user.getSalt()));
+        return userDao.add(user);
+    }
```

Comment:
I think maybe better will be setSalt and encrypt password in register method, but not sure